File: backend/src/main/kotlin/de/iteratec/konfi/rest/dto/ConferenceDto.java
package de.iteratec.konfi.rest.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModelProperty;
import io.swagger.annotations.ApiParam;

import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;

public class ConferenceDto {
    
    private Long id;

    @NotNull
    @Size(min = 3, max = 20)
    @ApiModelProperty(example = "KotlinKonf", required = true)
    private String name;

    @NotNull
    @Positive
    @ApiParam(example = "200")
    @ApiModelProperty(example = "200", required = true)
    private int maxAttendees;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm")
    @ApiModelProperty(example = "2022-06-04 09:00", required = true)
    private LocalDateTime startDate;

    @NotNull
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm")
    @ApiModelProperty(example = "2022-06-06 15:00", required = true)
    private LocalDateTime endDate;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getMaxAttendees() {
        return maxAttendees;
    }

    public void setMaxAttendees(int maxAttendees) {
        this.maxAttendees = maxAttendees;
    }

    public LocalDateTime getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDateTime startDate) {
        this.startDate = startDate;
    }

    public LocalDateTime getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDateTime endDate) {
        this.endDate = endDate;
    }

    @AssertTrue
    public Boolean checkDates() {
        return endDate.isAfter(startDate);
    }
}

File: backend/src/main/java/de/iteratec/konfi/rest/dto/ConferenceDto.kt
package de.iteratec.konfi.rest.dto

import com.fasterxml.jackson.annotation.JsonFormat
import io.swagger.annotations.ApiModelProperty
import io.swagger.annotations.ApiParam
import java.time.LocalDateTime
import javax.validation.constraints.AssertTrue
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

class ConferenceDto {
    var id: Long? = null

    @field:NotNull
    @field:ApiModelProperty(example = "KotlinKonf", required = true)
    @field:Size(min = 3, max = 20)
    var name: String? = null

    @field:NotNull
    @field:ApiParam(example = "200")
    @field:ApiModelProperty(example = "200", required = true)
    var maxAttendees = 0

    @field:NotNull
    @field:JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm")
    @field:ApiModelProperty(example = "2022-06-04 09:00", required = true)
    var startDate: LocalDateTime? = null

    @field:NotNull
    @field:JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm")
    @field:ApiModelProperty(example = "2022-06-06 15:00", required = true)
    var endDate: LocalDateTime? = null

    @AssertTrue
    fun checkDates(): Boolean {
        return endDate!!.isAfter(startDate)
    }
}